### PR TITLE
complete first draft

### PR DESCRIPTION
**Pull request by holyseo**

### CODE REVIEW

File: src/components/CommentList.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const CommentList = () => {
       setComments(comments);
       setIsLoading(false);
     });
-  }, []);
+  }, [comments]);
```
Comment (nc-modou) [inline src/components/CommentList.jsx]: How come you've added comments to this useEffect dependency array?
Remember that when something in the dependency array chances, it triggers another request. Looks like this is what is causing your repeating requests

**Review by nc-modou — COMMENTED** [review]
image
Your page seems to be making requests automatically... I left the site open for a while and the site made 800+ requests to your server to GET comments! Take a look at my comment to find out why this is happening.

Also, your comment post works smoothly and shows the user good feedback. However, it would be great for the user to see their comment reach the top of the comments so they can see that it's been posted. We can do this by using the response from the API when you post, and adding that new comment to the state.

Have a go implementing these fixes, then please resubmit a PR!

File: src/components/CommentList.jsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ const CommentList = () => {
       setComments(comments);
       setIsLoading(false);
     });
-  }, [comments]);
+  }, [updateComments]);
```
Comment (ghost) [inline src/components/CommentList.jsx:24]: This is far from ideal, I made 75 requests to your database just writing out a comment, this re renders your comments every time you type a character...

**Review by ghost — CHANGES_REQUESTED** [review]
This refactor has the same issue as before, while the loop is not infinite, I absolutely should _not_ be making a database request for every character in the post comment form. Please take a look into this and resubmit.

**Review by ghost — APPROVED** [review]
Much better. Comments are posting as they should. I would urge you to consider your colour pallete as it is highly inaccessible at this point

**Review by nc-poppy-m — APPROVED** [review]
Good change in comment palette, the articles list palette should also be changed but you can leave that for now as this was already approved by Verity. Merge when ready

**Pull request merged**